addons/acodec/mp3.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * Allegro5 MP3 reader.
+ * Requires MiniMP3 from https://github.com/lieff/minimp3
+ * author: Mark Watkin (pmprog) 2019
+ */
+
+
+#include "allegro5/allegro.h"
+#include "allegro5/allegro_acodec.h"
+#include "allegro5/allegro_audio.h"
+#include "allegro5/internal/aintern.h"
+#include "allegro5/internal/aintern_audio.h"
+#include "allegro5/internal/aintern_exitfunc.h"
+#include "allegro5/internal/aintern_system.h"
+#include "acodec.h"
+#include "helper.h"
+
+#define MINIMP3_IMPLEMENTATION
+#include <minimp3.h>
+#include <minimp3_ex.h>
+
+ALLEGRO_DEBUG_CHANNEL("acodec")
+
+static bool mp3_libinit = false;
+static mp3dec_t mp3d;
+
+void mp3_initminimp3();
```

Comment:
This should be `mp3_initminimp3(void)`. If you omit the `void` in C it means that the argument list is unspecified (unlike in C++).